sqlglot/dialects/tsql.py

```diff
@@ -539,3 +615,35 @@ def returning_sql(self, expression: exp.Returning) -> str:
             into = self.sql(expression, "into")
             into = self.seg(f"INTO {into}") if into else ""
             return f"{self.seg('OUTPUT')} {self.expressions(expression, flat=True)}{into}"
+
+        def transaction_sql(self, expression: exp.Transaction) -> str:
+            this = self.sql(expression, "this")
+            this = f" {this}" if this else ""
+            mark = expression.args.get("mark")
+            mark = f" WITH MARK {mark}" if mark else ""
+            return f"BEGIN TRANSACTION{this}{mark}"
+
+        def _durability_sql(self, expression) -> str:
+            durability = expression.args.get("durability")
+            durability_sql = ""
+            if durability is not None:
+                if durability:
+                    durability = "ON"
+                else:
+                    durability = "OFF"
+                durability_sql = f" WITH (DELAYED_DURABILITY = {durability})"
+            return durability_sql
+
+        def commit_sql(self, expression: exp.Commit) -> str:
+            this = self.sql(expression, "this")
+            this = f" {this}" if this else ""
+            transaction = expression.args.get("transaction")
+            transaction = f" {transaction}" if transaction else ""
+            return f"COMMIT{transaction}{this}{self._durability_sql(expression)}"
+
+        def rollback_sql(self, expression: exp.Rollback) -> str:
+            this = self.sql(expression, "this")
+            this = f" {this}" if this else ""
+            transaction = expression.args.get("transaction")
```

Comment:
```suggestion
            transaction = self.sql(expression, "transaction")
```

Reply:
Ok, replaced the two instances of `expression.args.get("transaction")`